### DISCUSSION TITLE
Local ParallelIO Build/Link with E3SM

### DISCUSSION THREAD

**Discussion by ackirby88** [E3SM-Project/E3SM#6150] [E3SM model help]
Hi E3SM Team,  
I've been able to build E3SM (maint-2.1) on the NCAR/NWSC Derecho system successfully and was able to run the test case `XS_2x5_ndays`. However, I had a few issues with linking a locally-installed version of ParallelIO.  

Inside `cime/CIME/build_scripts/buildlib.pio`, the comment states:
```python
    # If variable PIO_VERSION_MAJOR is defined in the environment then
    # we assume that PIO is installed on the system
    # and expect to find
    # PIO_LIBDIR, PIO_INCDIR, PIO_TYPENAME_VALID_VALUES
    # also defined in the environment.  In this case we
    # will use the installed pio and not build it here.
```

I tried setting these four variables via my own `.bashrc` file, as well as in my `run_e3sm.derecho.sh` file via _readonly_ variables. When setting the `do_case_build=true`, the compilation of e3sm.exe fails because it cannot find `use pio` for the Fortran files, as well as the all of the c-based functions used from the ParallelIO library. To get around this, I made symbolic links to the `libpioc.a` and `libpiof.a` static libraries in the `{TEST_CASE}/build/<compiler>/mct/lib` directory.  

**I was wondering the proper location for setting these `PIO` environment variables to get E3SM to pick up the local installation correctly?** Thank you very much in advance!

**Comment by jayeshkrishna** [comment]
We haven't tried E3SM builds with an installation of PIO outside the build directory (CESM might have this as a supported configuration).

**Reply by ackirby88** [reply]
Thank you for the feedback.  
I'll try adding a few lines in my `run_e3sm.derecho.sh` script to make the symbolic links automatically before the `case.build` operation.  
Please feel free to close this discussion if you believe this issue is resolved.

**Comment by xylar** [comment]
This does seem like a worthwhile thing to fix.  We also build SCORPIO externally in a shared space for our Compass and Polaris packages, so it would be convenient if we could point Omega to these instead of redundantly building SCORPIO again each time we build Omega itself.  It's not a major problem for us, just a matter of convenience.

cc @philipwjones @grnydawn
